pkg/kubectlcmd/client.go

```diff
@@ -102,6 +102,103 @@ func (c *Client) Apply(dryRun, verbose bool, kubeconfig, context, namespace stri
 	return stdout.String(), csError, nil
 }
 
+// GetAll runs the kubectl get all with the provided arguments
+func (c *Client) GetAll(dryRun bool, kubeconfig, context, namespace string, extraArgs ...string) (string, string, error) {
+	args := []string{"get", "all"}
+	if kubeconfig != "" {
+		args = append(args, "--kubeconfig", kubeconfig)
+	}
+	if context != "" {
+		args = append(args, "--context", context)
+	}
+	if namespace != "" {
+		args = append(args, "-n", namespace)
+	}
+	args = append(args, extraArgs...)
+
+	cmd := exec.Command("kubectl", args...)
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	cmdStr := strings.Join(cmd.Args, " ")
+	if dryRun {
+		logAndPrint("GetAll is in dry run mode, would be running in the following namespace %s:\n%s\n", namespace, cmdStr)
```

Comment:
i think getall should work normally in dryrun - just don't prune the objects.

Reply:
Nice catch. Fixed.